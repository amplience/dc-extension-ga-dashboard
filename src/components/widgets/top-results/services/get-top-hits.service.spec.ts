import getTopHits, { getTopHitsForCsv } from './get-top-hits.service';
import type { SearchIndex } from 'dc-management-sdk-js';
import topHitsFixture from '../__fixtures__/top-hits-single-item-list';

describe('getTopHits service', () => {
  it('should list top hits for a given search index and params', async () => {
    const mockedGetItems = jest.fn().mockImplementation(() => topHitsFixture);
    const mockedTopResultsGet = jest.fn().mockImplementation(() => {
      return {
        getItems: mockedGetItems,
      };
    });
    const mockedSearchIndex = {
      related: {
        'top-hits': {
          get: mockedTopResultsGet,
        },
      },
    };
    const params = {
      limit: 5,
      includeReplicas: true,
      startDate: '2020-01-01',
      endDate: '2020-01-01',
    };

    const result = await getTopHits(
      (mockedSearchIndex as unknown) as SearchIndex,
      params
    );

    expect(mockedTopResultsGet).toHaveBeenCalledWith({
      clickAnalytics: true,
      ...params,
    });
    expect(mockedGetItems).toHaveBeenCalled();
    expect(result).toMatchSnapshot();
  });
});

describe('getTopHitsForCsv service', () => {
  it('should list csv formatted top hits for a given search index', async () => {
    const mockedGetItems = jest.fn().mockImplementation(() => topHitsFixture);
    const mockedTopResultsGet = jest.fn().mockImplementation(() => {
      return {
        getItems: mockedGetItems,
      };
    });
    const mockedSearchIndex = {
      related: {
        'top-hits': {
          get: mockedTopResultsGet,
        },
      },
    };
    const params = {
      includeReplicas: true,
      startDate: '2020-01-01',
      endDate: '2020-01-01',
    };

    const result = await getTopHitsForCsv(
      (mockedSearchIndex as unknown) as SearchIndex,
      params
    );

    expect(result).toMatchSnapshot();
  });
});
