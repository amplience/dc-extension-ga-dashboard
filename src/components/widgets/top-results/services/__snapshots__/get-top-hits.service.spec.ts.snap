// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTopHits service should list top hits for a given search index and params 1`] = `
Array [
  Object {
    "count": 17,
    "hit": "db1f8fe3-392f-4427-b070-ecc7b35b31b1",
  },
]
`;

exports[`getTopHitsForCsv service should list csv formatted top hits for a given search index 1`] = `
Array [
  Object {
    "contentItemId": "db1f8fe3-392f-4427-b070-ecc7b35b31b1",
    "count": 17,
  },
]
`;
